### ISSUE TITLE
Keybinding conflicts occurred in Template Preference Page

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext-eclipse#1361]
1. start runtime eclipse
2. open preferences and go to e.g. Xtend templates
3. close Preference Dialog
4. open the Preference dialog again

this will result in

```
Binding(F2,
	ParameterizedCommand(Command(org.eclipse.ui.edit.rename,Rename,
		Rename the selected item,
		Category(org.eclipse.ui.category.file,File,null,true),
		org.eclipse.ui.internal.WorkbenchHandlerServiceHandler@65b1dc67,
		,,true),null),
	org.eclipse.ui.defaultAcceleratorConfiguration,
	org.eclipse.ui.contexts.window,,,system)
Binding(F2,
	ParameterizedCommand(Command(org.eclipse.ui.edit.text.showInformation,Show Tooltip Description,
		Displays information for the current caret location in a focused hover,
		Category(org.eclipse.ui.category.textEditor,Text Editing,Text Editing Commands,true),
		org.eclipse.ui.internal.WorkbenchHandlerServiceHandler@c129b51,
		,,true),null),
	org.eclipse.ui.defaultAcceleratorConfiguration,
	org.eclipse.xtext.ui.embeddedTextEditorScope,,,system)
```

on the first call into
`org.eclipse.xtext.ui.editor.embedded.EmbeddedEditorActions.initialize()`

the active contexts are
`[org.eclipse.ui.contexts.dialogAndWindow, org.eclipse.ui.contexts.dialog]`
on the second

`[org.eclipse.ui.contexts.window, org.eclipse.jdt.ui.A_OpenActionSet, org.eclipse.jdt.ui.JavaActionSet, org.eclipse.jdt.ui.JavaElementCreationActionSet, org.eclipse.ui.edit.text.actionSet.presentation, org.eclipse.ui.contexts.dialogAndWindow, org.eclipse.ui.cheatsheets.actionSet, org.eclipse.jdt.ui.CodingActionSet, org.eclipse.ui.edit.text.actionSet.navigation, org.eclipse.ui.actionSet.openFiles, org.eclipse.ui.externaltools.ExternalToolsSet, org.eclipse.jdt.ui.SearchActionSet, org.eclipse.ui.actionSet.keyBindings, org.eclipse.ui.edit.text.actionSet.convertLineDelimitersTo, org.eclipse.debug.ui.launchActionSet, org.eclipse.jdt.junit.JUnitActionSet, org.eclipse.ui.NavigateActionSet, org.eclipse.ui.edit.text.actionSet.annotationNavigation, org.eclipse.search.searchActionSet, org.eclipse.text.quicksearch.actionSet, org.eclipse.jdt.debug.ui.JDTDebugActionSet, org.eclipse.jdt.ui.text.java.actionSet.presentation, org.eclipse.debug.ui.breakpointActionSet]
`

so it looks like the contexts for the workbench window dont get dectivated properly or not yet. but why is this no problem in the first call.

=> i dont know if this is an Xtext or a Platform issue cc @iloveeclipse
see also https://bugs.eclipse.org/bugs/show_bug.cgi?id=539409

**Comment by cdietrich** [comment]
i also wonder why we need the `org.eclipse.xtext.ui.embeddedTextEditorScope` at all.
i also assume that a ShellListener (on activate) wont help if the embedded editor is in a view

**Comment by cdietrich** [comment]
created https://bugs.eclipse.org/bugs/show_bug.cgi?id=560762

**Comment by cdietrich** [comment]
another quesiton is: the default editor in the preference (preview) is readonly. 
@szarnekow do you know any custom bindings for the embedded editor that should work there?

do we need a possibiliy to create an embedded editor with no bindings/actions?

**Comment by cdietrich** [comment]
Fixed in 2.22
